packages/@pollyjs/adapter-node-http/src/-private/transport-wrapper.js

```diff
@@ -0,0 +1,290 @@
+import http from 'http';
+import NodeUrl from 'url';
+
+import isObjectLike from 'lodash-es/isObjectLike';
+import { URL } from '@pollyjs/utils';
+
+import isBinaryBuffer from '../utils/is-binary-buffer';
+import isContentEncoded from '../utils/is-content-encoded';
+import mergeChunks from '../utils/merge-chunks';
+
+const nativeRequestMapping = new WeakMap();
+
+const { keys } = Object;
+
+export default class TransportWrapper {
+  constructor(transport, { name, adapter }) {
+    this.name = name;
+    this.adapter = adapter;
+    this.transport = transport;
+  }
+
+  isPatched() {
+    return nativeRequestMapping.has(this.transport);
+  }
+
+  patch() {
+    // make sure it's not already patched
+    this.adapter.assert(
+      `The ${
+        this.name
+      } transport has already been patched, please stop any running Polly instances`,
+      !this.isPatched()
+    );
+
+    nativeRequestMapping.set(this.transport, {
+      request: this.transport.request,
+      get: this.transport.get
+    });
+
+    this.transport.request = this.createRequestWrapper();
+
+    // In Node 10+, http.get no longer references http.request by the export
+    // so we need to make sure we wrap it as well.
+    // https://github.com/nodejs/node/blob/v10.0.0/lib/https.js#L275
+    this.transport.get = this.createGetWrapper();
+  }
+
+  restore() {
+    this.adapter.assert(
+      `Cannot restore unpatched transport ${this.name}`,
+      this.isPatched()
+    );
+
+    const { request, get } = nativeRequestMapping.get(this.transport);
+
+    this.transport.request = request;
+    this.transport.get = get;
+
+    nativeRequestMapping.delete(this.transport);
+  }
+
+  getBodyFromChunks(chunks, headers) {
+    const { adapter } = this;
+
+    //  If we have headers and there is content-encoding it means that
+    //  the body shouldn't be merged but instead persisted as an array
+    //  of hex strings so that the responses can be mocked one by one.
```

Comment:
Here is my take on an alternative (took me a few minutes to understand what this meant):

```js
//  If content-encoding is set in the header then the body/content
//  should not be concatenated.  Instead, the chunks should
//  be preserved as-is so that each chunk can be mocked individually
```